react/components/CostCentersAutocomplete.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react'
+import { useQuery } from 'react-apollo'
+import { AutocompleteInput } from 'vtex.styleguide'
+import { useIntl } from 'react-intl'
+
+import { messages } from './customers-admin'
+import GET_COST from '../queries/costCentersByOrg.gql'
+
+interface Props {
+  onChange: (value: { value: string | null; label: string }) => void
+  organizationId?: string
+}
+
+const initialState = {
+  search: '',
+  page: 1,
+  pageSize: 25,
+  sortOrder: 'ASC',
+  sortedBy: 'name',
+}
+
+const CostCenterAutocomplete = ({ onChange, organizationId }: Props) => {
+  const { formatMessage } = useIntl()
+  const [costCenterTextInput, setCostCenterTextInput] = useState('')
+  const [debouncedSearchTerm, setDebouncedSearchTerm] =
+    useState(costCenterTextInput)
+
+  const { data, loading, refetch } = useQuery(GET_COST, {
+    variables: {
+      ...initialState,
+      id: organizationId,
+    },
+    ssr: false,
+    notifyOnNetworkStatusChange: true,
+    skip: !organizationId,
+  })
+
+  const onClear = () => {
+    setCostCenterTextInput('')
+    onChange({ value: null, label: '' })
+  }
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedSearchTerm(costCenterTextInput)
+    }, 500) // 500ms delay
```

Comment:
should we make this a constant? maybe something like `SEARCH_TERM_DELAY_MS`? (then we can remove the `// 500ms delay` comment as well).

If we define this in a constant file we can even reuse for the other file (`OrganizationsAutocomplete.tsx`)

Reply:
I don't think it's necessary, since this can be changed depending on the strategy, the delay can change and also for this repo there are not just 2 autocomplete

Reply:
I`ve added that suggestion